src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -307,25 +309,37 @@ static void OnGetAddrInfo(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t i
     {
         VerifyOrReturn(sdCtx->HasAddress(), sdCtx->Finalize(kDNSServiceErr_BadState));
 
-        if (domainName.compare(kOpenThreadDot) == 0)
+        if (domainName.compare(kSrpDot) == 0)
```

Comment:
I think this is broken.  You can get multiple SRP results, then a non-SRP result with !kDNSServiceFlagsMoreComing, and this code will end up sitting and waiting for 250ms when it shouldn't.  I think there's some confusion here about what kDNSServiceFlagsMoreComing means.  Which is understandable, because it can be confusing.  If set, all it means is "something on this same connection has more data to deliver right now".  In our case, it could be either of the two resolves, any of the GetAddrInfo those kicked off, etc.  In particular, it does not mean "no more data coming for the args that got passed here"; it means "no more stuff I know about right this second for anything that is using this connection".